src/lib/support/CommonPersistentData.h

```diff
@@ -0,0 +1,83 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Contains a class handling creation of linked list of stored persistent data.
+ */
+
+#include <lib/support/PersistentData.h>
+
+namespace chip {
+
+typedef uint8_t entryIndex;
+constexpr entryIndex kUndefinedEntryIndex = 0;
+
+struct StoredDataList : public PersistentData<kPersistentBufferMax>
+{
+    static constexpr TLV::Tag TagFirstEntry() { return TLV::ContextTag(1); }
+    static constexpr TLV::Tag TagEntryCount() { return TLV::ContextTag(2); }
+
+    chip::entryIndex first_entry = kUndefinedEntryIndex;
+    uint8_t entry_count          = 0;
+
+    StoredDataList() = default;
+    StoredDataList(chip::entryIndex first) : first_entry(first), entry_count(1) {}
+
+    CHIP_ERROR UpdateKey(DefaultStorageKeyAllocator & key) override
+    {
+        key.StoredDataList();
```

Comment:
What does this do? Is this calling a function `char *(void)` and if yes why? The naming of this method overlaps with the struct name .. I find this confusing.

Reply:
There is indeed a name conflict, that was fixed by changing the name to FabricList in DefaultStorageKeyAllocator.h